### PR TITLE
admin can now delete comments

### PR DESCRIPTION
**Pull request by o-bm**
- updated whoami route to fetch the role of the user. returns json username & role. nothing needs to be changed
- added logic for rendering solution page based on role of user. admin gets to see a different page than a normal user
- admin can reply to comments and have a button delete to delete that comment. 

image

### CODE REVIEW

**Comment by o-bm** [comment]
close #58

File: client/src/components/RoleBasedSolution.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from 'react';
+import Solution from '../pages/Solution.tsx';
+import AdminSolution from '../pages/AdminSolution.tsx';
+
+const RoleBasedSolution = () => {
+  const [isAdmin, setIsAdmin] = useState(false);
+
+  //fetch the username from the server
+  useEffect(() => {
+    fetch("/api/users/whoami")
```
Comment (arthurhohinng) [inline client/src/components/RoleBasedSolution.tsx:10]: does whoami return the role as well? I thought it only returned the utorid?

File: server/middleware/AuthenticationMiddleware.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ async function authMiddleware(req, res, next) {
         user = await db.User.create({
           username: utorid,
           email: http_mail,
-          role: "user",
+          role: "admin",
```
Comment (arthurhohinng) [inline server/middleware/AuthenticationMiddleware.js:15]: is this defaulting to creating an admin? If it is we need to change it

File: server/controllers/UserController.js [hunk 3]
```diff
@@ -2,9 +2,28 @@ const db = require("../models/index");
 const User = db.User;
 //change
 exports.getMe = async (req, res) => {
-  res.status(200).json({ username: req.headers.utorid });
+  try {
```
Comment (arthurhohinng) [inline server/controllers/UserController.js:5]: oh I see we changed it to return role

File: client/src/pages/AdminSolution.tsx
Comment (arthurhohinng) [inline client/src/pages/AdminSolution.tsx:1]: Can we make the delete button within the comment div?

**Review by arthurhohinng — COMMENTED** [review]
Couple comments to address
Also I prefer commit message to be something more like:
Add admin comment deletion... for example

In our case though something that would fit better from our prev feature commits would be:
Admin Comment Deletion

File: client/src/components/RoleBasedSolution.tsx (hunk 1 above)
Comment (o-bm) [reply client/src/components/RoleBasedSolution.tsx]: I updated it to return the role too

File: client/src/pages/AdminSolution.tsx
Comment (o-bm) [reply client/src/pages/AdminSolution.tsx:1]: for sure

**Pull request merged**